src/pybind/mgr/dashboard/controllers/rbd.py

```diff
@@ -453,3 +465,81 @@ def _clone(ioctx):
             return _rbd_call(child_pool_name, _clone)
 
         return _rbd_call(pool_name, _parent_clone)
+
+
+@ApiController('/block/image/trash')
+class RbdTrash(RESTController):
+    RESOURCE_ID = "pool_name/image_id"
+
+    @ViewCache()
+    def _rbd_pool_list(self, pool_name):
+        rbd_inst = rbd.RBD()
+        with mgr.rados.open_ioctx(pool_name) as ioctx:
+            images = rbd_inst.trash_list(ioctx)
+            result = []
+            for trash in images:
+                trash['pool_name'] = pool_name
+                trash['deletion_time'] = "{}Z".format(trash['deletion_time'].isoformat())
+                trash['deferment_end_time'] = "{}Z".format(trash['deferment_end_time'].isoformat())
+                result.append(trash)
+            return result
+
+    def _trash_list(self, pool_name=None):
```

Comment:
Although they were identical, the _rbd_pool_list call was referencing different methods.
I have renamed _rbd_pool_list  to _trash_pool_list, so we can better differentiate them.

Refactoring this would require I added parameters to the method and conditional calls inside, do you think it is worth it?